src/parsing/convertXDMReturnValue.ts

```diff
@@ -257,13 +257,12 @@ export default function convertXDMReturnValue<
 							resolve(getNextResult())
 						).catch((error) => {
 							printAndRethrowError(expression, error);
-							throw error;
 						}),
 				};
 			} else {
-				toReturn = {
+				toReturn = ({
 					next: () => new Promise((resolve) => resolve(getNextResult())),
-				} as AsyncIterableIterator<any>;
+				} as unknown) as AsyncIterableIterator<any>;
```

Comment:
This forced cast seems to be because toReturn is not actually an `AsyncIterableIterator` but an `AsyncIterator`. The "Iterable" version needs a [Symbol.asyncIterator] property.

Reply:
Yep, but at the moment of initial implementation, AsyncIterableIterator was not supported in all browsers, not willing to change this behaviour right now in order to let this PR snowball any further